src/extensions/agenda.py

```diff
@@ -0,0 +1,193 @@
+import arc
+import hikari
+import aiohttp
+from urllib.parse import urlparse
+import datetime
+
+from src.utils import role_mention, hedgedoc_login
+from src.hooks import restrict_to_channels, restrict_to_roles
+from src.config import CHANNEL_IDS, ROLE_IDS, UID_MAPS, AGENDA_TEMPLATE_URL
+
+
+plugin = arc.GatewayPlugin(name="Agenda")
+
+agenda = plugin.include_slash_group("agenda", "Interact with the agenda.")
+
+
+def generate_date_choices() -> list[str]:
+    """Generate date options for the next 7 days."""
+    today = datetime.date.today()
```

Comment:
These generated day options will never update, so `today` will always be the day blockbot was started on. The only way around this is using day names instead of dates, or generating options with autocomplete (such as something I've done [here](https://github.com/novanai/timetable-sync/blob/master/bot/extensions/timetable.py#L18-L67)).